integration-libs/opf/base/core/facade/opf-payment.service.ts

```diff
@@ -45,4 +186,56 @@ export class OpfPaymentService implements OpfPaymentFacade {
       paymentVerificationPayload,
     });
   }
+
+  submitPayment(submitInput: SubmitInput): Observable<boolean> {
+    return this.submitPaymentCommand.execute({ submitInput });
+  }
+
+  protected displayError(error: PaymentError | undefined): void {
+    this.globalMessageService.add(
+      {
+        key: error?.message ? error.message : this.defaultError.message,
+      },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  }
+
+  protected handleBadRequestError(errorType: string): string {
+    let message = this.defaultError.message;
+    switch (errorType) {
+      case 'EXPIRED':
+        message = 'opf.payment.errors.cardExpired';
+        break;
+      case 'INSUFFICENT_FUNDS':
```

Comment:
Does it make sense here to create an enum with such values in the model file? Just for sake potential re-use in different places. It's always better also from IDE perspective to see possible options.

Reply:
done, btw payment errors topic is still in discussion with Richard/bamboo. 
It looks like we might only need to handle level1 errors (success, rejected and cancel). Those above authorization errors belong to level2, better to keep them for now. We'll remove them when confirmed by Richard/Bamboo.